### PR TITLE
Comment fixes for CRM/Utils directory

### PR DESCRIPTION
**Pull request by saurabhbatra96**
Changed loads of comments to follow the Doxygen commenting standards. Removed empty comments. Found an empty function that takes no parameters and returns no values.

### CODE REVIEW

File: CRM/Utils/ReCAPTCHA.php [hunk 1]
```diff
@@ -63,9 +61,8 @@ public static function &singleton() {
     return self::$_singleton;
   }
 
-  /**
-   */
   public function __construct() {
+    // Empty function?
```
Comment (saurabhbatra96) [inline CRM/Utils/ReCAPTCHA.php]: @eileenmcnaughton Couldn't decide if this was a bug or a feature. :-)

File: CRM/Utils/API/MatchOption.php [hunk 2]
```diff
@@ -126,7 +125,7 @@ public function fromApiInput($apiRequest) {
           break;
 
         default:
-          // be forgiveful of sloppily api calls
+          // be forgiving of sloppy api calls
```
Comment (saurabhbatra96) [inline CRM/Utils/API/MatchOption.php]: :-)

Comment (totten) [inline CRM/Utils/API/MatchOption.php]: This change is the cherry on top of a whipped-cream commit.

Comment (totten) [inline CRM/Utils/API/MatchOption.php]: In digging, it looks like I put in that line. Can't remember if it was willfully ironic or just incidental.

Let's call it willful...

Comment (saurabhbatra96) [inline CRM/Utils/API/MatchOption.php]: When I read the line I just couldn't resist changing it :-).

**Comment by colemanw** [comment]
Hmm, I'm not sure about `{@inheritDoc}` vs `@inheritDoc` (or `{@inheritdoc}` or `@inheritdoc`).
I don't find any consensus on the internet about which one is "right"; everyone seems to have their own opinion.
Currently CiviCRM is at least consistent with itself (we use `@inheritDoc` everwhere). This PR proposes to change this in some places but not others, which would leave us internally inconsistent.

**Comment by saurabhbatra96** [comment]
I was following these guidelines: https://www.drupal.org/node/1354#inheritdoc
But I concede that your point regarding consistency is very much valid. I'll revert back on this change in the next commit.

File: CRM/Utils/ReCAPTCHA.php (hunk 1 above)
Comment (eileenmcnaughton) [inline CRM/Utils/ReCAPTCHA.php]: yes!! why would we have that!

**Pull request merged**